src/import-data/ImportData.ts

```diff
@@ -1,72 +1,88 @@
 import _ from 'lodash/fp';
-import { Fragment, useCallback, useState } from 'react';
-import { h } from 'react-hyperscript-helpers';
+import { Fragment, ReactNode, useState } from 'react';
+import { div, h } from 'react-hyperscript-helpers';
 import { spinnerOverlay } from 'src/components/common';
 import { Ajax } from 'src/libs/ajax';
 import { Dataset } from 'src/libs/ajax/Catalog';
 import { resolveWdsUrl, WdsDataTableProvider } from 'src/libs/ajax/data-table-providers/WdsDataTableProvider';
+import colors from 'src/libs/colors';
 import { withErrorReporting } from 'src/libs/error';
 import Events, { extractWorkspaceDetails } from 'src/libs/events';
 import * as Nav from 'src/libs/nav';
 import { notify } from 'src/libs/notifications';
 import { useOnMount } from 'src/libs/react-utils';
 import { asyncImportJobStore } from 'src/libs/state';
 import * as Utils from 'src/libs/utils';
+import { WorkspaceInfo } from 'src/libs/workspace-utils';
 import { useDataCatalog } from 'src/pages/library/dataBrowser-utils';
 import { notifyDataImportProgress } from 'src/workspace-data/import-jobs';
 
-import { TemplateWorkspaceInfo } from './import-types';
+import {
+  BagItImportRequest,
+  CatalogDatasetImportRequest,
+  CatalogSnapshotsImportRequest,
+  EntitiesImportRequest,
+  ImportRequest,
+  PFBImportRequest,
+  TDRSnapshotExportImportRequest,
+  TDRSnapshotReferenceImportRequest,
+  TemplateWorkspaceInfo,
+} from './import-types';
 import { ImportDataDestination } from './ImportDataDestination';
 import { ImportDataOverview } from './ImportDataOverview';
 import { isProtectedSource } from './protected-data-utils';
+import { useImportRequest } from './useImportRequest';
 
-// ImportData handles all the information relating to the page itself - this includes:
-// * Reading from the URL
-// * Loading initial Data
-// * Managing the import
-export const ImportData = () => {
+const getTitleForImportRequest = (importRequest: ImportRequest): string => {
+  if (importRequest.type === 'tdr-snapshot-export') {
+    return `Importing snapshot ${importRequest.snapshotName}`;
+  }
+
+  if (
+    importRequest.type === 'tdr-snapshot-reference' ||
+    importRequest.type === 'catalog-dataset' ||
+    importRequest.type === 'catalog-snapshots'
+  ) {
+    return 'Linking data to a workspace';
+  }
+
+  return 'Importing data to a workspace';
+};
+
+export interface ImportDataProps {
+  importRequest: ImportRequest;
+}
```

Comment:
Why use an interface that contains only an `ImportRequest` instead of just the `ImportRequest` by itself?

Reply:
The interface is defining the props for a React Component: ImportData. Props is always an object. This component currently only takes one prop: an ImportRequest.
https://react.dev/learn/passing-props-to-a-component